src/Form/DropdownField.jsx

```diff
@@ -16,17 +16,43 @@ export const DropdownField = ({
 }) => {
   const [field, meta, helpers] = useField(props)
 
-  const handleChange = useCallback(v => helpers.setValue(v), [helpers])
-
-  const selectedOption = useMemo(
-    () =>
-      options.find(option => field.value === option[optionAttributes.value]),
-    [field, options, optionAttributes]
+  const handleChange = useCallback(
+    (option, v) => {
+      if (multiple) {
+        const value = [...field.value]
+        const idx = value.indexOf(option.id ?? option)
+        if (idx >= 0) {
+          value.splice(idx, 1)
+        } else {
+          value.push(option)
```

Comment:
You have to decide on `option` or `option.id`. You can't test for option.id and push the entire option object. For example:

```
const initialValue = ['a', b']
const options = [{id:'a', label:'The A element'},  { id:'b', label: 'The B element'}, {id:'c', label: 'The C element'}]
// I am checking the C element and my resulting (after submit) value will be:
const finalValue = ['a', 'b', {id: 'c', label: 'The C element'}]